SU2_CFD/include/solvers/CFVMFlowSolverBase.hpp

```diff
@@ -0,0 +1,1217 @@
+/*!
+ * \file CFVMFlowSolverBase.hpp
+ * \brief Base class template for all FVM flow solvers.
+ * \version 7.0.6 "Blackbird"
+ *
+ * SU2 Project Website: https://su2code.github.io
+ *
+ * The SU2 Project is maintained by the SU2 Foundation
+ * (http://su2foundation.org)
+ *
+ * Copyright 2012-2020, SU2 Contributors (cf. AUTHORS.md)
+ *
+ * SU2 is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * SU2 is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with SU2. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include "CSolver.hpp"
+#include "../../../Common/include/omp_structure.hpp"
+#include "../../../Common/include/toolboxes/geometry_toolbox.hpp"
+
+
+template<class VariableType, ENUM_REGIME FlowRegime>
+class CFVMFlowSolverBase : public CSolver {
```

Comment:
Declarations and inline function in the cpp file.
The first template parameter is the type for `nodes` this is so we can still have de-virtualization.